### ISSUE TITLE
Model loss/total loss

### ISSUE THREAD

**Issue by seovchinnikov** [argman/EAST#90]
Hey!
I'm trying to fugure out why
model with pretrained vgg weights  always demonstrates starting training  model loss of 0.1 and total loss (as I understood its just an addition of regularization loss) of 0.4 on my own dataset. Further, model loss remains the same and total loss converges to model loss of 0.1.
But after training the model produces messy results.
![img_1](https://user-images.githubusercontent.com/20027569/35008820-e4f06a9a-fb0e-11e7-8bbb-d68573795101.jpg)

**Comment by argman** [comment]
i dont know, maybe your input and output cannot make the network converge

**Comment by LouisRuan** [comment]
Your initial loss is close to 0.01? I think the loss function has any questions.

**Comment by YanShuang17** [comment]
I meet the same problem, when I train the model using my own dataset, the model loss is always 0.01:
```
Step 000000, model loss 0.0100, total loss 0.0424, 1.29 seconds/step, 10.82 examples/second
2018-05-06 10:10:37.596315: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 1794 get requests, put_count=1424 evicted_count=1000 eviction_rate=0.702247 and unsatisfied allocation rate=0.819398
2018-05-06 10:10:37.596423: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 100 to 110
Step 000010, model loss 0.0100, total loss 0.0407, 1.51 seconds/step, 9.25 examples/second
2018-05-06 10:10:51.571265: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 1432 get requests, put_count=1566 evicted_count=1000 eviction_rate=0.63857 and unsatisfied allocation rate=0.62081
2018-05-06 10:10:51.571379: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 256 to 281
Step 000020, model loss 0.0100, total loss 0.0390, 0.72 seconds/step, 19.40 examples/second
Step 000030, model loss 0.0100, total loss 0.0372, 0.72 seconds/step, 19.46 examples/second
Step 000040, model loss 0.0100, total loss 0.0355, 0.71 seconds/step, 19.60 examples/second
Step 000050, model loss 0.0100, total loss 0.0339, 0.71 seconds/step, 19.63 examples/second
Step 000060, model loss 0.0100, total loss 0.0324, 0.71 seconds/step, 19.59 examples/second
Step 000070, model loss 0.0100, total loss 0.0309, 0.71 seconds/step, 19.63 examples/second
2018-05-06 10:11:30.291204: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 12405 get requests, put_count=12493 evicted_count=1000 eviction_rate=0.0800448 and unsatisfied allocation rate=0.0782749
2018-05-06 10:11:30.291236: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 655 to 720
Step 000080, model loss 0.0100, total loss 0.0296, 0.73 seconds/step, 19.22 examples/second
Step 000090, model loss 0.0100, total loss 0.0284, 0.72 seconds/step, 19.41 examples/second
Step 000100, model loss 0.0100, total loss 0.0272, 0.72 seconds/step, 19.32 examples/second
Step 000110, model loss 0.0100, total loss 0.0260, 0.78 seconds/step, 17.91 examples/second
Step 000120, model loss 0.0100, total loss 0.0251, 0.72 seconds/step, 19.54 examples/second
Step 000130, model loss 0.0100, total loss 0.0242, 0.72 seconds/step, 19.53 examples/second
Step 000140, model loss 0.0100, total loss 0.0233, 0.72 seconds/step, 19.56 examples/second
Step 000150, model loss 0.0100, total loss 0.0225, 0.71 seconds/step, 19.59 examples/second
Step 000160, model loss 0.0100, total loss 0.0218, 0.72 seconds/step, 19.47 examples/second
Step 000170, model loss 0.0100, total loss 0.0212, 0.72 seconds/step, 19.52 examples/second
Step 000180, model loss 0.0100, total loss 0.0205, 0.72 seconds/step, 19.51 examples/second
Step 000190, model loss 0.0100, total loss 0.0200, 0.72 seconds/step, 19.44 examples/second
Step 000200, model loss 0.0100, total loss 0.0194, 0.73 seconds/step, 19.17 examples/second
...
...
Step 009900, model loss 0.0100, total loss 0.0100, 0.73 seconds/step, 19.14 examples/second
Step 009910, model loss 0.0100, total loss 0.0100, 0.83 seconds/step, 16.88 examples/second
Step 009920, model loss 0.0100, total loss 0.0100, 0.75 seconds/step, 18.67 examples/second
Step 009930, model loss 0.0100, total loss 0.0100, 0.74 seconds/step, 18.96 examples/second
Step 009940, model loss 0.0100, total loss 0.0100, 0.75 seconds/step, 18.76 examples/second
Step 009950, model loss 0.0100, total loss 0.0100, 0.75 seconds/step, 18.75 examples/second
Step 009960, model loss 0.0100, total loss 0.0100, 0.73 seconds/step, 19.11 examples/second
Step 009970, model loss 0.0100, total loss 0.0100, 0.73 seconds/step, 19.09 examples/second
Step 009980, model loss 0.0100, total loss 0.0100, 0.74 seconds/step, 18.86 examples/second
Step 009990, model loss 0.0100, total loss 0.0100, 0.74 seconds/step, 18.81 examples/second
```
But when using the icdar2015 dataset, the model loss is normal and can get a good result.
So what the dataset should be like? any limits?

**Comment by IngleJaya95** [comment]
@seovchinnikov Facing Same issues, any solution/suggestion?

**Comment by JackeyDeng** [comment]
I met the same issue too, does anyone knows the reason?

**Comment by PGrothaus** [comment]
This seems to be caused when no valid training examples are in the dataset.
It occured to me when using '###' as text label for all training examples. Those examples are ignored in the loss. (Same for '*'):

In `icdar.py`:
```python
if label == '*' or label == '###':
                text_tags.append(True)
            else:
                text_tags.append(False)
```
and later on those examples are masked out:
```python
if tag:
     cv2.fillPoly(training_mask, poly.astype(np.int32)[np.newaxis, :, :], 0)

```

I'd suggest to check the training data set.

**Comment by jiangmiaotong** [comment]
@

> This seems to be caused when no valid training examples are in the dataset.
> It occured to me when using '###' as text label for all training examples. Those examples are ignored in the loss. (Same for '*'):
> 
> In `icdar.py`:
> 
> ```python
> if label == '*' or label == '###':
>                 text_tags.append(True)
>             else:
>                 text_tags.append(False)
> ```
> 
> and later on those examples are masked out:
> 
> ```python
> if tag:
>      cv2.fillPoly(training_mask, poly.astype(np.int32)[np.newaxis, :, :], 0)
> ```
> 
> I'd suggest to check the training data set.

But when I trained on  one image and the same at each time , sometimes it appears to be 0.01,sometimes it works.

**Comment by peter-peng-w** [comment]
Any update or solution of this problem? I am also encountering with the problem of model loss fixing on 0.01.

**Comment by aiFTong** [comment]
> fixing

https://github.com/argman/EAST/issues/90#issuecomment-492505506

**Comment by sdzbft** [comment]
> Any update or solution of this problem? I am also encountering with the problem of model loss fixing on 0.01.

 text_tags.append(False)  set False
